### PR TITLE
Main/SDL - Add troubleshooting keybinds/functions

### PR DESCRIPTION
**Pull request by Optimoos**
The emulator logic is now separated from the loop in main. This allows for keybinds to perform troubleshooting functions. H toggles a complete pause, when paused J will do a single frame. L toggles an arbitrary delay to be better able to manually monitor memory or the debug output.

### CODE REVIEW

**Comment by sophiajt** [comment]
These will be helpful for debugging.

Can you also print some info to the console (or maybe even the screen) to tell the user they've hit one of these keys and what the state is?

**Comment by Optimoos** [comment]
I've been working on getting some additional output going in-program instead of console only, but that has become another rabbit hole as SDL doesn't handle text on its own. I tried getting Dear IMGUI functional in Jakt, but as usual I'm in over my head. Not sure if there's a better / preferred library to focus on. Maybe Qt given Andreas' background with it? Of course, that will be an even bigger rabbit hole than IMGUI.

Anyway, for now console output is in.

**Comment by sophiajt** [comment]
It'd be best to keep it all SDL, to keep the requirements as small as possible. We can leave that for later. I'm just thinking of something simple like drawing a little box and writing a message into it.

**Comment by sophiajt** [comment]
I was playing around with the emulator this morning and noticed that this PR slows it down quite a bit. Any ideas?

File: src/main.jakt [hunk 1]
```diff
@@ -25,101 +25,111 @@ function main(args: [String]) {
...
-                                        joypad.press_a()
-                                    }
-                                    X => {
-                                        joypad.press_b()
-                                    }
-                                    D => {
-                                        system.debug_enable()
-                                    }
+            loop {
+                mut event = sdl.poll_event()
```
Comment (sophiajt) [inline src/main.jakt:36]: Oh, this might be it. I'll do some research but we shouldn't check for events every pixel, only every frame.

Comment (sophiajt) [reply src/main.jakt:36]: Yeah that seems to be it. Working on a fix

**Pull request merged**